tests/js/keyboardInputTests.js

```diff
@@ -113,12 +114,28 @@ https://github.com/gpii/universal/LICENSE.txt
         jqUnit.assertEquals(msg, expected, keyboardInput.container.hasClass(className));
     };
 
+    gpii.tests.firstDiscovery.keyboardInput.checkKeypress = function (expected, actual) {
+        jqUnit.assertEquals("keypress character should be \"" + expected + "\"", expected, actual);
+    };
+
+    gpii.tests.firstDiscovery.keyboardInput.checkUserInput = function (keyboardInput, expected) {
+        // Check both the model value and the HTML input value
+        jqUnit.assertEquals("userInput should be \"" + expected + "\"",
+                            expected, keyboardInput.model.userInput);
+        jqUnit.assertEquals("HTML input value should be \"" + expected + "\"",
+                            expected, keyboardInput.container.val());
+    };
+
     gpii.tests.firstDiscovery.keyboardInput.setUpTooltipTest = function (keyboardInput) {
         // Set the focus and tooltip state to a known starting point
-        $("#gpiic-tests-other-input").focus();
+        gpii.tests.firstDiscovery.keyboardInput.focusOther();
         keyboardInput.tooltip.close();
     };
 
+    gpii.tests.firstDiscovery.keyboardInput.focusOther = function () {
+        $("#gpiic-tests-other-input").focus();
```

Comment:
We should use the fluid.focus utility here and in the fixtures below (implemented in FluidDocument.js) unless we can verify that in all of our supported configurations of IE, the behaviour has gone away described in the comment above the implementation there in which this will sometimes be honoured asynchronously.